### ISSUE TITLE
Jenkins build broken

### ISSUE THREAD

**Issue by Flamefire** [Return-To-The-Roots/s25client#1700]
Currently the build on Jenkins is broken on linux preventing the next nightly which would also fix the updater bug.

```
      Start 16: Test_integration
      Start 19: Test_lua
 1/25 Test #16: Test_integration .................***Failed    0.05 sec
Running 131 test cases...
unknown location(0): [4;31;49mfatal error: in "AI/PlayerHasBld_IsCorrect": memory access violation at address: 0x557400000000: no mapping at fault address[0;39;49m
/source/tests/s25Main/./worldFixtures/WorldFixture.h(108): [1;36;49mlast checkpoint[0;39;49m

[1;31;49m*** 1 failure is detected in the test module "RTTR_Integration"
[0;39;49m

      Start 25: Test_UI
 2/25 Test #19: Test_lua .........................***Failed    0.05 sec
double free or corruption (out)
Running 27 test cases...
unknown location(0): [4;31;49mfatal error: in "LuaTestSuite/LuaEqual_IsCorrect": signal: SIGABRT (application abort requested)[0;39;49m
/source/tests/s25Main/lua/testLua.cpp(37): [1;36;49mlast checkpoint: "LuaEqual_IsCorrect" fixture dtor[0;39;49m
double free or corruption (out)
unknown location(0): [4;31;49mfatal error: in "LuaTestSuite/AssertionThrows": signal: SIGABRT (application abort requested)[0;39;49m
/source/tests/s25Main/lua/testLua.cpp(46): [1;36;49mlast checkpoint: "AssertionThrows" fixture dtor[0;39;49m
double free or corruption (out)
unknown location(0): [4;31;49mfatal error: in "LuaTestSuite/ScriptLoading": signal: SIGABRT (application abort requested)[0;39;49m
/source/tests/s25Main/lua/testLua.cpp(52): [1;36;49mlast checkpoint: "ScriptLoading" fixture dtor[0;39;49m
double free or corruption (out)
unknown location(0): [4;31;49mfatal error: in "LuaTestSuite/BaseFunctions": signal: SIGABRT (application abort requested)[0;39;49m
/source/tests/s25Main/lua/testLua.cpp(85): [1;36;49mlast checkpoint: "BaseFunctions" fixture dtor[0;39;49m
double free or corruption (out)
unknown location(0): [4;31;49mfatal error: in "LuaTestSuite/Translations": signal: SIGABRT (application abort requested)[0;39;49m
/source/tests/s25Main/lua/testLua.cpp(119): [1;36;49mlast checkpoint: "Translations" fixture dtor[0;39;49m
unknown location(0): [4;31;49mfatal error: in "LuaTestSuite/GameFunctions": memory access violation at address: 0x7ffd00000000: no mapping at fault address[0;39;49m
/source/tests/s25Main/lua/GameWithLuaAccess.h(99): [1;36;49mlast checkpoint[0;39;49m

[1;31;49m*** 6 failures are detected in the test module "RTTR_Lua"
```

This doesn't happen on any other CI, not locally on Linux 2 systems and also not using the docker container locally:

```
docker run -it -e USER --rm -v /home/alex/git/s25client:/source:ro -v /tmp/result:/result:rw --workdir /tmp/build  registry.mytrap.de/rttr/cross-compiler/linux/linux-amd64-docker:master /bin/bash 
root@2bd4558292f5:/tmp/build# export JENKINS_SERVER_COOKIE=1
# Insert `architecture="linux.x86_64"` to build.sh
root@2bd4558292f5:/tmp/build# /source/tools/ci/jenkins/build.sh 
```

So I think it is something on the build server. Can you investigate  please @Flow86? Maybe clearing the ccache already solves this.

**Comment by stefson** [comment]
seems more like compile is working, but tests are broken?

**Comment by Flow86** [comment]
I've deleted the local cache. lets see if that helps. (or do we run into an out of memory? the build containers are quiet limited, but I don't think the tests take more than 8gig ram, do we?)

**Comment by Flamefire** [comment]
It yields memory access violations so I don't think it is an OOM issue.
I tried to build locally with ASAN on Linux und VLD on Windows and didn't find any issues.

When trying the build in the docker container by hand (not using the build.sh, but IMO same commands) I got an error that the static boost libs were not build with -fPIC. Using the build script the issue vanished. Very strange, not sure if related
